channeldb/node.go

```diff
@@ -0,0 +1,175 @@
+package channeldb
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+var (
+	// nodeAnnouncementBucket stores announcement config pertaining to node.
+	// This bucket allows one to query for persisted node announcement
+	// config and use it when starting orrestarting a node
+	nodeAnnouncementBucket = []byte("nab")
+)
+
+type NodeAnnouncement struct {
+	// Alias indicate the human readable name that a node operator can
+	// assign to their node for better readability and easier identification
+	Alias string
+
+	// Color represent the hexadecimal value that node operators can assign
+	// to their nodes. It's represented as a hex string.
+	Color string
+
+	// IdentityPub is the node's current identity public key. Any
+	// channel/topology related information received by this node MUST be
+	// signed by this public key.
+	IdentityPub *btcec.PublicKey
+
+	db *NodeAnnouncementDB
+}
+
+type NodeAnnouncementDB struct {
+	backend kvdb.Backend
+}
+
```

Comment:
we already persist our node announcement in the bucket storing all the collected node announcement. Can we not just grab our latest one from there?

Reply:
I don't know where the collected node announcements are stored. Are you referring to [here](https://github.com/lightningnetwork/lnd/blob/613bfc07fb664086e78e8c58c9e1c8785e4656d9/channeldb/graph.go#L3847) where all nodes within a channel graph are stored and their node announcements (if they have any)?.

Reply:
yes - we store our node announcement there too right? if so, then i dont think we need a whole new bucket for it. Let me know if my assumption is incorrect

Reply:
@ellemouton  I explored the code related to graph DB, and it turns out you're right. We're indeed storing our node along with the node announcement, and we can easily retrieve the stored values from this [method](https://github.com/lightningnetwork/lnd/blob/98c52df4d2378512b721f4e1140efc5ac0ef5904/channeldb/graph.go#L750)  when lnd starts.

Since this is the case, I will close this PR and open a new one to replace it. Does that make sense?